### ISSUE TITLE
Did anyone get reasonable performance on KITTI?

### ISSUE THREAD

**Issue by yzou2** [dontLoveBugs/DORN_pytorch#15]
Thanks for sharing the code. But when I train it on KITTI (considering sparse ground truth only), the delta_1 is just 0.15 after 15 epochs. This is much worse than the original paper. Did anyone produce reasonable results (maybe not as good as paper, but delta_1 > 0.85)?

**Comment by angshine** [comment]
> Thanks for sharing the code. But when I train it on KITTI (considering sparse ground truth only), the delta_1 is just 0.15 after 15 epochs. This is much worse than the original paper. Did anyone produce reasonable results (maybe not as good as paper, but delta_1 > 0.85)?

I can get `d1=0.894, d2=0.980, dd31=0.998` with dense ground truth, but haven't tried on the sparse ground truth only case.

**Comment by yzou2** [comment]
> > Thanks for sharing the code. But when I train it on KITTI (considering sparse ground truth only), the delta_1 is just 0.15 after 15 epochs. This is much worse than the original paper. Did anyone produce reasonable results (maybe not as good as paper, but delta_1 > 0.85)?
> 
> I can get `d1=0.894, d2=0.980, dd31=0.998` with dense ground truth, but haven't tried on the sparse ground truth only case.

How did you get the dense ground truth for KITTI?

**Comment by angshine** [comment]
> > > Thanks for sharing the code. But when I train it on KITTI (considering sparse ground truth only), the delta_1 is just 0.15 after 15 epochs. This is much worse than the original paper. Did anyone produce reasonable results (maybe not as good as paper, but delta_1 > 0.85)?
> > 
> > 
> > I can get `d1=0.894, d2=0.980, dd31=0.998` with dense ground truth, but haven't tried on the sparse ground truth only case.
> 
> How did you get the dense ground truth for KITTI? I only have the sparse points transformed from LIDAR.

When generating the training data, set `interp=True` while using the `generate_depth_map()` function. This will generate dense depth map interpolated from the sparse ground truth from lidar. I think you can also use other depth completion methods [(kitti_depth_completion)](http://www.cvlibs.net/datasets/kitti/eval_depth.php?benchmark=depth_completion) to get dense lidar point cloud, and project the point cloud to image, then using interpolation method to get dense depth map.

**Comment by olivemm** [comment]
hello，did you solve the problem？ I haven't got the reasonable result on KITTI. Could you please tell me how did you solve it?  Did the dense groundtruth make sense?

**Comment by LCJHust** [comment]
> > Thanks for sharing the code. But when I train it on KITTI (considering sparse ground truth only), the delta_1 is just 0.15 after 15 epochs. This is much worse than the original paper. Did anyone produce reasonable results (maybe not as good as paper, but delta_1 > 0.85)?
> 
> I can get `d1=0.894, d2=0.980, dd31=0.998` with dense ground truth, but haven't tried on the sparse ground truth only case.

Hi, did you modify something on the auhor's code to get these excellent results except using dense depth map? I mean, the network, loss function or parameters? I can't get considerable results with author‘s code on KITTI

**Comment by chenyilun95** [comment]
@LCJHust Hi, did you get the correct result now ? I've tried trained only on only one image but the result on that image is still low (Delta1=0.15 at most).

**Comment by angshine** [comment]
> > Thanks for sharing the code. But when I train it on KITTI (considering sparse ground truth only), the delta_1 is just 0.15 after 15 epochs. This is much worse than the original paper. Did anyone produce reasonable results (maybe not as good as paper, but delta_1 > 0.85)?
> 
> I can get `d1=0.894, d2=0.980, dd31=0.998` with dense ground truth, but haven't tried on the sparse ground truth only case.

Sorry for the late reply. I couldn't get acceptable result when I used the original code of this repo, so I basically reproduced the DORN based on this repo. However, I did this work during an internship so I can't publish the code. I think the main training issue is the dead ReLU, and the hyperparameter is hard to tune. In the end, I change the optimizer to LAMB (LARS trick on ADAM optimizer), and the training is much more smooth and I can get the result mentioned above. (However, I didn't control all the variables when I re-implement the code. So I suggest everyone who stuck at the training of DORN could try plugging in the LAMB optimizer and give it a try.)
